src/integrations/integr_chrome.rs

```diff
@@ -224,141 +289,325 @@ async fn setup_chrome_session(
                     }
+                } {
+                    Ok(html) => {
+                        format!("innerHtml of {}:\n\n{}", tab_lock.state_string(), html)
+                    },
+                    Err(e) => {
+                        format!("can't fetch innerHtml of {}: {}", tab_lock.state_string(), e.to_string())
+                    },
                 }
-            },
-            Command::Click(args) => {
-                let (tab, open_tab_log) = session_open_tab(chrome_session, &args.tab_id).await?;
-                tool_log.push(open_tab_log);
-                let content = click_on_point(&tab, &args.point).await?;
-                tool_log.push(content);
-            },
-            Command::InsertText(args) => {
-                let (tab, open_tab_log) = session_open_tab(chrome_session, &args.tab_id).await?;
-                tool_log.push(open_tab_log);
-                let content = insert_text(&tab, &args.text).await?;
-                tool_log.push(content);
-            },
+            };
+            tool_log.push(log);
+        },
+        Command::Reload(args) => {
+            let tab = {
+                let mut chrome_session_locked = chrome_session.lock().await;
+                let chrome_session = chrome_session_locked.as_any_mut().downcast_mut::<ChromeSession>().ok_or("Failed to downcast to ChromeSession")?;
+                session_get_tab_arc(chrome_session, &args.tab_id).await?
+            };
+            let log = {
+                let tab_lock = tab.lock().await;
+                let chrome_tab = tab_lock.headless_tab.clone();
+                match chrome_tab.reload(false, None) {
+                    Ok(_) => {
+                        format!("reload of {} successful", tab_lock.state_string())
+                    },
+                    Err(e) => {
+                        format!("reload of {} failed: {}", tab_lock.state_string(), e.to_string())
+                    },
+                }
+            };
+            tool_log.push(log);
+        },
+        Command::ClickAt(args) => {
+            let tab = {
+                let mut chrome_session_locked = chrome_session.lock().await;
+                let chrome_session = chrome_session_locked.as_any_mut().downcast_mut::<ChromeSession>().ok_or("Failed to downcast to ChromeSession")?;
+                session_get_tab_arc(chrome_session, &args.tab_id).await?
+            };
+            let log = {
+                let tab_lock = tab.lock().await;
+                match {
+                    let mapped_point = Point {
+                        x: args.point.x / tab_lock.screenshot_scale_factor,
+                        y: args.point.y / tab_lock.screenshot_scale_factor,
+                    };
+                    tab_lock.headless_tab.click_point(mapped_point).map_err(|e| e.to_string())?;
+                    tab_lock.headless_tab.wait_until_navigated().map_err(|e| e.to_string())?;
```

Comment:
will exit the whole function

Reply:
tested, works well
it stores an error in the tool_log, all ok